**Talent acquisition strategy:** <br>Identify the academic backgrounds and departments most interested in AI for targeted recruitment.

In [2]:
import pandas as pd
import plotly.express as px

In [3]:
data = pd.read_csv('/content/Cumulative_Activity_Form_Responses.csv')

**Data processing and exploration**

In [4]:
data.head()

Your Full Name  \
0      Student 1   
1      Student 2   
2      Student 3   
3      Student 4   
4      Student 5   

  Are you an undergraduate, hold a Bachelor's, or a Master's degree?  \
0                                         Bachelor's                   
1                                            Masters                   
2                                      Undergraduate                   
3                                         Bachelor's                   
4                                         Bachelor's                   

  Which academic department are you currently affiliated with?  \
0                                    Web Development             
1                                   Computer Science             
2                                    Web Development             
3                               Software Development             
4                                          Marketing             

                 Why are you learning generative AI?  \
0  to help me develop strategies that can help me...   
1  creating models that can produce new data samp...   
2         All about gemerayive AI and how to use AI    
3                                Help me with my job   
4                         something about our future   

  Do you need online career support from us?  \
0                                        Yes   
1                                        Yes   
2                                        Yes   
3                                        Yes   
4                                         No   

  Do you require online learning support in AI?  
0                                           Yes  
1                                            No  
2                                           Yes  
3                                           Yes  
4                                           Yes

In [5]:
data.shape

(23, 6)

In [6]:
# Copy the data
df_general = data.copy()

# Rename the columns for easy access
df_general.columns = ['Name', 'Degree', 'Department', 'Motivation', 'Career_Support', 'Learning_Support']

df_general.head(2)

Name      Degree        Department  \
0  Student 1  Bachelor's   Web Development   
1  Student 2     Masters  Computer Science   

                                          Motivation Career_Support  \
0  to help me develop strategies that can help me...            Yes   
1  creating models that can produce new data samp...            Yes   

  Learning_Support  
0              Yes  
1               No

In [44]:
# Filter rows where individuals require online learning support in AI
df = df_general[df_general.Learning_Support == 'Yes']
df.shape[0]

15

In [45]:
# Count the number of interested individuals per degree and department
df = df.groupby(['Degree', 'Department']).size().reset_index(name='Count')
df

Degree              Department  Count
0      Bachelor's     Business Management      2
1      Bachelor's               Marketing      2
2      Bachelor's    Software Development      1
3      Bachelor's         Web Development      1
4         Masters  Information Technology      1
5         Masters               Marketing      1
6         Masters    Software Development      1
7   Undergraduate     Business Management      1
8   Undergraduate        Computer Science      1
9   Undergraduate  Information Technology      2
10  Undergraduate         Web Development      2

In [46]:
# Sort the data by count in descending order
df = df.sort_values(by='Count', ascending=False)
df

Degree              Department  Count
0      Bachelor's     Business Management      2
1      Bachelor's               Marketing      2
9   Undergraduate  Information Technology      2
10  Undergraduate         Web Development      2
2      Bachelor's    Software Development      1
3      Bachelor's         Web Development      1
4         Masters  Information Technology      1
5         Masters               Marketing      1
6         Masters    Software Development      1
7   Undergraduate     Business Management      1
8   Undergraduate        Computer Science      1

**Data visualization**

In [47]:
# Horizontal Bar Chart
fig = px.bar(df, x='Count', y='Degree', color='Department',
             title='Interest in AI Learning Support by Degree and Department',
             labels={'Count':'Number of Individuals', 'Degree':'Academic Degree'},
             color_continuous_scale=px.colors.sequential.Plasma,
             template='plotly_dark')
fig.show()

In [48]:
# Donut Chart
fig = px.pie(df, values='Count', names='Degree',
             title='Distribution of Interest in AI Learning Support by Degree',
             hole=.3,
             color_discrete_sequence=px.colors.sequential.Plasma,
             template='plotly_dark')
fig.show()

**Insights**

**The data results indicate:**

**Undergraduate students** from **Information Technology and Web Development** departments show an **equal interest in AI**, with a count of 2 respectively.

**Bachelor's degree** holders from **Business Management and Marketing departments** also demonstrate a **similar interest in AI**, each with a count of 2.

Only one Masters student each from Information Technology, Marketing, and Software Development departments are interested in AI.

One Undergraduate student each from Business Management and Computer Science departments show an interest in AI.

These results suggest that the recruitment strategy should **primarily target** **Undergraduate students in Information Technology and Web Development** departments, followed by Bachelor's degree holders in Business Management and Marketing departments.

**Identification of the most popular academic departments among the respondents.**

In [50]:
df_general

Name         Degree              Department  \
0    Student 1     Bachelor's         Web Development   
1    Student 2        Masters        Computer Science   
2    Student 3  Undergraduate         Web Development   
3    Student 4     Bachelor's    Software Development   
4    Student 5     Bachelor's               Marketing   
5    Student 6        Masters  Information Technology   
6    Student 7  Undergraduate        Computer Science   
7    Student 8     Bachelor's               Marketing   
8    Student 9        Masters    Software Development   
9   Student 10  Undergraduate         Web Development   
10  Student 11        Masters     Business Management   
11  Student 12  Undergraduate  Information Technology   
12  Student 13  Undergraduate  Information Technology   
13  Student 14        Masters    Software Development   
14  Student 15  Undergraduate     Business Management   
15  Student 16     Bachelor's    Software Development   
16  Student 17     Bachelor's     Business Management   
17  Student 18        Masters         Web Development   
18  Student 19  Undergraduate        Computer Science   
19  Student 20     Bachelor's        Computer Science   
20  Student 21        Masters               Marketing   
21  Student 22     Bachelor's     Business Management   
22  Student 23  Undergraduate  Information Technology   

                                           Motivation Career_Support  \
0   to help me develop strategies that can help me...            Yes   
1   creating models that can produce new data samp...            Yes   
2          All about gemerayive AI and how to use AI             Yes   
3                                 Help me with my job            Yes   
4                          something about our future             No   
5                         Utilize generative AI tools            Yes   
6                       Explore AI work opportunities            Yes   
7                             Work with Google Gemini            Yes   
8    How to find a job job with emerging tehcnologies             No   
9                         Integrate AI with my career            Yes   
10                      Find the right career pathway            Yes   
11                How to use AI to leverage my career            Yes   
12                                     Grow my career             No   
13                                     Find a new job             No   
14                         Enhance my teaching skills            Yes   
15                    Keep updated on AI advancements            Yes   
16     Enhance my knowledge in data analysis using AI            Yes   
17        Upgrade my career to be able to work withAI             No   
18  I want to have full knowledge about how the ai...            Yes   
19   Understand how it works and apply it to finance              No   
20      To use AI tools in different business aspects             No   
21                             How to work on ChatGPT            Yes   
22         Be able to clearly understand how Ai works             No   

   Learning_Support  
0               Yes  
1                No  
2               Yes  
3               Yes  
4               Yes  
5               Yes  
6                No  
7               Yes  
8               Yes  
9               Yes  
10               No  
11              Yes  
12              Yes  
13               No  
14              Yes  
15               No  
16              Yes  
17               No  
18              Yes  
19               No  
20              Yes  
21              Yes  
22               No

In [52]:
df_popular = df_general['Department'].value_counts().reset_index()
df_popular

Department  count
0         Web Development      4
1        Computer Science      4
2    Software Development      4
3  Information Technology      4
4     Business Management      4
5               Marketing      3

**Data visualization**

In [55]:
fig = px.bar(df_popular,
             x='count',
             y='Department',
             title='Most Popular Academic Departments Among Respondents',
             labels={'count':'Number of Respondents',
                     'Department':'Department'},
             color='Department',
             template='plotly_dark')

fig.show()

**Insights**

**The analytics task reveals the following:**

Among the respondents, the most popular academic departments are **Web Development, Computer Science, Software Development, Information Technology, and Business Management**, each with a Respondent Count of 4.
These departments are followed closely by Marketing, which has a Respondent Count of 3.
It's worth noting that **there is no single stand-out department**; the top five are equally popular.
The data does not show a clear preference for either technical (i.e., Web Development, Computer Science, Software Development, Information Technology) or business-oriented (i.e., Business Management, Marketing) departments.
Further analysis or more data might be needed to identify any significant differences or trends among the respondents' academic department choices.

**Analysis of the proportion of respondents requiring online career and learning support in AI.**

In [57]:
df_general.head(1)

Name      Degree       Department  \
0  Student 1  Bachelor's  Web Development   

                                          Motivation Career_Support  \
0  to help me develop strategies that can help me...            Yes   

  Learning_Support  
0              Yes

In [64]:
# Calculate proportions
df_career_support = df_general['Career_Support'].value_counts(normalize=True).reset_index()
df_learning_support = df_general['Learning_Support'].value_counts(normalize=True).reset_index()


In [65]:
df_career_support

Career_Support  proportion
0            Yes    0.652174
1             No    0.347826

In [63]:
df_learning_support

Learning_Support  proportion
0              Yes    0.652174
1               No    0.347826

In [69]:
# Create bar chart
fig = px.bar(df_career_support, y='Career_Support', x='proportion',
             color='Career_Support',
             title='Proportion of respondents requiring online career support',
             labels={'proportion':'Proportion of respondents',
                     'Career_Support': 'Need Online Career Support'},
             template='plotly_dark')
fig.show()

In [70]:
# Create donut chart
fig = px.pie(df_career_support, values='proportion', names='Career_Support',
             title='Proportional distribution of respondents requiring online career support',
             template='plotly_dark', hole=0.3)
fig.show()

In [68]:
# Create horizontal bar chart for learning support
fig = px.bar(df_learning_support, y='Learning_Support', x='proportion',
             color='Learning_Support',
             title='Proportion of respondents requiring online learning support in AI',
             labels={'proportion':'Proportion of respondents',
                     'Learning_Support': 'Require Online Learning Support in AI'},
             template='plotly_dark')
fig.show()

In [71]:
# Create donut chart
fig = px.pie(df_learning_support, values='proportion', names='Learning_Support',
             title='Proportional distribution of respondents requiring online learning support in AI',
             template='plotly_dark', hole=0.3)
fig.show()

The analytics task reveals that a significant majority of respondents expressed a need for online career and learning support in AI.

Approximately **65.2% **of respondents indicated a **requirement for online learning and career support**.
In contrast, only about 34.8% of participants responded negatively to the same inquiry.
Therefore, there's a **substantial demand** for online resources that **cater to AI learning and career development**.
The data suggests an opportunity to enhance online support systems or introduce new, comprehensive platforms that directly address this need.

**Determination of the most common level of educational attainment among respondents interested in learning generative AI.**

In [10]:
# filter data by respondents interested in learning generative AI
df_intrested_AI = df_general[~df_general['Motivation'].isnull()]
df_intrested_AI.shape[0]

23

In [12]:
df_intrested_AI['Motivation'].value_counts().sort_values(ascending=False)

Motivation
to help me develop strategies that can help me in my career                                                  1
creating models that can produce new data samples that resemble the training data they were provided with    1
All about gemerayive AI and how to use AI                                                                    1
Help me with my job                                                                                          1
something about our future                                                                                   1
Utilize generative AI tools                                                                                  1
Explore AI work opportunities                                                                                1
Work with Google Gemini                                                                                      1
How to find a job job with emerging tehcnologies                                                     

In [14]:
# determine the most common level of educational attainment
df_education_and_motivation_AI = df_intrested_AI['Degree'].value_counts().reset_index()

df_education_and_motivation_AI

Degree  count
0     Bachelor's      8
1  Undergraduate      8
2        Masters      7

**Data visualization**

In [16]:
fig = px.bar(df_education_and_motivation_AI,
             x='count',
             y='Degree',
             color='Degree',
             title='Educational Attainment of AI Learners',
             labels={'count':'Number of Respondents',
                     'Degree':'Educational Level'},
             color_discrete_sequence=px.colors.qualitative.Pastel,
             template='plotly_dark')
fig.show()

**Insights **

**The analysis of the educational level of respondents interested in learning generative AI reveals the following:**

The **most common level of educational** attainment is a tie between those with a **Bachelor's degree** and those who are **Undergraduates**, each category having 8 respondents.
The next common level of educational attainment is a Master's degree with 7 respondents.
This suggests that the interest in generative AI is spread across different stages of higher education, with a slight emphasis on the undergraduate level and those who have completed a Bachelor's degree.